include/boost/unordered/unordered_node_map.hpp

```diff
@@ -106,10 +123,11 @@ namespace boost {
           }
           BOOST_CATCH(...)
           {
-            boost::allocator_deallocate(al,
-              boost::pointer_traits<
-                typename boost::allocator_pointer<A>::type>::pointer_to(*p->p),
-              1);
+            using pointer_type=typename boost::allocator_pointer<A>::type;
+            using pointer_traits=boost::pointer_traits<pointer_type>;
+
+            boost::allocator_deallocate(
+              al,pointer_traits::pointer_to(*(p->p)),1);
```

Comment:
Why not just `boost::allocator_deallocate(al,p->p,1)`?

Reply:
In our cases, the `element_type` stores a `value_type*` which we can't deallocate through if the Allocator is using fancy pointers.

I think it's easiest for us to store `value_type*` and convert from raw pointers to fancy pointers only in these certain instances.

Reply:
Yes, I agree. Thank you!